contracts/finance/payment-splitter/traits.rs

```diff
@@ -0,0 +1,196 @@
+use brush::{
+    declare_storage_trait,
+    traits::{
+        AccountId,
+        AccountIdExt,
+        Balance,
+        InkStorage,
+    },
+};
+use ink_prelude::{
+    vec::Vec,
+};
+use ink_storage::{
+    traits::SpreadLayout,
+    collections::HashMap as StorageHashMap,
+    Vec as StorageVec,
+};
+pub use payment_splitter_derive::PaymentSplitterStorage;
+
+use brush::traits::ZERO_ADDRESS;
+#[cfg(feature = "std")]
+use ink_storage::traits::StorageLayout;
+
+#[derive(Default, Debug, SpreadLayout)]
+#[cfg_attr(feature = "std", derive(StorageLayout))]
+pub struct PaymentSplitterData {
+    pub total_shares: Balance,
+    pub total_released: Balance,
+    pub shares: StorageHashMap<AccountId, Balance>,
+    pub released: StorageHashMap<AccountId, Balance>,
+    pub payees: StorageVec<AccountId>,
+}
+
+declare_storage_trait!(PaymentSplitterStorage, PaymentSplitterData);
+
+#[derive(strum_macros::AsRefStr)]
+pub enum PaymentSplitterError {
+    LengthMismatch,
+    NoPayees,
+    AccountHasNoShares,
+    AccountIsNotDuePayment,
+    AccountZeroAddress,
+    SharesAreZero,
+    AlreadyHasShares,
+    TransferFailed,
+}
+
+/// This contract allows to split native token payments among a group of accounts. The sender does not need to be aware
+/// that the native token will be split in this way, since it is handled transparently by the contract.
+///
+/// The split can be in equal parts or in any other arbitrary proportion. The way this is specified is by assigning each
+/// account to a number of shares. Of all the native tokens that this contract receives, each account will then be able to claim
+/// an amount proportional to the percentage of total shares they were assigned.
+///
+/// PaymentSplitter `follows a _pull payment_ model. This means that payments are not automatically forwarded to the
+/// accounts but kept in this contract, and the actual transfer is triggered as a separate step by calling the `release`
+/// function.
+///
+/// ** Note **: In the substrate balance of contract decreases each block. Because it pays rent for the storage.
+/// So during `release`, each next user will get fewer native tokens.
+///
+/// This module is used through embedding of `PaymentSplitterData` and implementation of `PaymentSplitter` and
+/// `PaymentSplitterStorage` traits.
+#[brush::trait_definition]
+pub trait PaymentSplitter: PaymentSplitterStorage {
+    /// Getter for the total shares held by payees.
+    #[ink(message)]
+    fn total_shares(&self) -> Balance {
+        self.get().total_shares.clone()
```

Comment:
isn't Balance `Copy`?

Reply:
It is, but I prefer explicitly call `clone`=)